x509-cert/tests/certificate.rs

```diff
@@ -116,7 +117,7 @@ fn reencode_cert() {
         include_bytes!("examples/026EDA6FA1EDFA8C253936C75B5EEBD954BFF452.fake.der");
     let defer_cert = DeferDecodeCertificate::from_der(der_encoded_cert).unwrap();
 
-    let parsed_tbs = TbsCertificate::from_der(defer_cert.tbs_certificate).unwrap();
+    let parsed_tbs = TbsCertificate::<Rfc5280>::from_der(defer_cert.tbs_certificate).unwrap();
```

Comment:
I don't get why I need to specify the profile here. I feel like I'm missing something.

Reply:
Default generic parameters aren't used for method invocations (or any expression). Inference is instead.

[`std` works around this by hardcoding the default parameter in certain contexts](https://doc.rust-lang.org/std/collections/struct.HashMap.html#impl-HashMap%3CK%2C%20V%2C%20RandomState%3E), although that works better for inherent methods than it does traits.

An alternative to a default generic parameter which might provide better ergonomics is to use type aliases instead.

Reply:
Yeah type aliases would work.